File: sorting/bubble-sort/bubble_sort.py
import numpy as np


def bubble_sort(arr):
    """ in-place array bubble sorting """
    size = len(arr)
    for i in range(size):
        stop = True
        for j in range(0, size-i-1):
            if arr[j] > arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]
                stop = False
        if stop:
            break
    return arr


if __name__ == '__main__':
    original = np.round(np.random.random(100) * 100).astype('int')
    sorted = bubble_sort(original.copy())
    print(f'original:\n\t{original}\n\nsorted:\n\t{sorted}\n')

File: sorting/bubble-sort/bubbleSort.js
/**
 * In-place array bubble-sorting
 * @param {Array} arr Array to sort
 * @returns {Array} sorted array
 */
function bubbleSort(arr) {
  const size = arr.length;
  for (let i = 0; i < size; i++) {
    let stop = true;
    for (let j = 0; j < size - i - 1; j++) {
      const a = arr[j];
      const b = arr[j + 1];
      if (a > b) {
        arr[j] = b;
        arr[j + 1] = a;
        stop = false;
      }
    }
    if (stop) {
      break;
    }
  }
  return arr;
}

// Test
const randomArray = [...new Array(100).keys()].map((x) => Math.round(Math.random() * 100));
console.log(`original:\n\t${JSON.stringify(randomArray)}\n\nsorted:\n\t${JSON.stringify(bubbleSort(randomArray))}\n`);
